sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -754,60 +742,28 @@ void gnrc_rpl_send_DAO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination, uint
     idx = gnrc_netif2_ipv6_addr_idx(netif, &dodag->dodag_id);
     me = &netif->ipv6.addrs[idx];
 
-    mutex_lock(&(gnrc_ipv6_fib_table.mtx_access));
-
-    /* add external and RPL FIB entries */
-    for (size_t i = 0; i < gnrc_ipv6_fib_table.size; ++i) {
-        fib_entry_t *fentry = &gnrc_ipv6_fib_table.data.entries[i];
-        if (fentry->lifetime != 0) {
-            if (!(fentry->next_hop_flags & FIB_FLAG_RPL_ROUTE)) {
-                ptr = &tmp;
-                if (!ext_processed) {
-                    DEBUG("RPL: Send DAO - building external transit\n");
-                    if ((tmp = _dao_transit_build(NULL, lifetime, true)) == NULL) {
-                        DEBUG("RPL: Send DAO - no space left in packet buffer\n");
-                        mutex_unlock(&(gnrc_ipv6_fib_table.mtx_access));
-                        return;
-                    }
-                    ext_processed = true;
-                }
-            }
-            else {
-                ptr = &pkt;
-                if (!int_processed) {
-                    DEBUG("RPL: Send DAO - building internal transit\n");
-                    if ((tr_int = pkt = _dao_transit_build(NULL, lifetime, false)) == NULL) {
-                        DEBUG("RPL: Send DAO - no space left in packet buffer\n");
-                        mutex_unlock(&(gnrc_ipv6_fib_table.mtx_access));
-                        return;
-                    }
-                    int_processed = true;
-                }
-            }
-            ipv6_addr_t *addr = (ipv6_addr_t *) fentry->global->address;
-            if (ipv6_addr_is_global(addr)) {
-                size_t prefix_length = (fentry->global_flags >> FIB_FLAG_NET_PREFIX_SHIFT);
-
-                DEBUG("RPL: Send DAO - building target %s/%d\n",
-                      ipv6_addr_to_str(addr_str, addr, sizeof(addr_str)),
-                      (int) prefix_length);
-
-                if ((*ptr = _dao_target_build(*ptr, addr, (uint8_t) prefix_length)) == NULL) {
-                    DEBUG("RPL: Send DAO - no space left in packet buffer\n");
-                    mutex_unlock(&(gnrc_ipv6_fib_table.mtx_access));
-                    return;
-                }
-            }
+    /* add external and RPL FT entries */
+    /* TODO: nib: dropped support for external transit options for now */
```

Comment:
What do you mean by that?

Reply:
In case we don't get this in before your vacation and someone else needs to take over: can you please answer my question?

Reply:
RPL is able to distribute routing states that were not learned/produced by RPL, but from an external source – like another routing protocol or a manually admnistered route entry. These routing states should be marked with the `E` flag in DAO transit optoins [1].
>   External (E): 1-bit flag.  The 'E' flag is set to indicate that the
         parent router redistributes external targets into the RPL
         network.  An external Target is a Target that has been learned
         through an alternate protocol.  The external targets are listed
         in the Target options that immediately precede the Transit
         Information option.  An external Target is not expected to
         support RPL messages and options.

In the "old" fib implementation user defined flags could be stored along the forwarding entry. We used that to mark a forwarding entry as "learned by RPL" and used this information to decide whether to set or not set the `E` flag in DAOs.

This feature is missing in the current `nib_ft` implementation that's why I put a note about dropped support there.

[1] https://tools.ietf.org/html/rfc6550#section-6.7.8

Reply:
Okay. I guess we'll need to extend the NIB for something like this later on ;-).